src/compiler/ast_native.cc

```diff
@@ -642,6 +642,11 @@ class ASTNativeCompilerImpl {
           "split_index"_a = node->split_index,
           "right_categories_flag"_a = right_categories_flag);
       }
+
+      oss << fmt::format(
+          "((data[{split_index}].fvalue >= 0) && !isinf(data[{split_index}].fvalue) &&"
+          "(fabsf(data[{split_index}].fvalue) <= (float)(1U << FLT_MANT_DIG)) && (",
```

Comment:
NAN has a special handling, since test nodes in Treelite have a default direction (`default_left`). But yes, INF handling can be folded into the range check.